streams/src/main/java/org/apache/kafka/streams/state/internals/KeyValueToTimestampedKeyValueByteStoreAdapter.java

```diff
@@ -127,9 +136,25 @@ public <R> QueryResult<R> query(
         final QueryConfig config) {
 
 
+
         final long start = config.isCollectExecutionInfo() ? System.nanoTime() : -1L;
-        final QueryResult<R> result = store.query(query, positionBound, config);
+        QueryResult<R> result = store.query(query, positionBound, config);
+
+        if (result.isSuccess()) {
+            if (query instanceof KeyQuery || query instanceof TimestampedKeyQuery) {
```

Comment:
We should add a comment above this line:
```
// this adapter always needs to return a `value-with-timestamp` result to hold up its contract
// thus, we need to add the dummy `-1` timestamp even for `KeyQuery` and `RangeQuery`
```

Reply:
ok